src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php

```diff
@@ -34,12 +34,17 @@ public function process(ContainerBuilder $container)
         $types = array();
 
         foreach ($container->findTaggedServiceIds('form.type') as $serviceId => $tag) {
-            $alias = isset($tag[0]['alias'])
-                ? $tag[0]['alias']
-                : $serviceId;
+            // The following if-else block is deprecated and can be removed
+            // in Symfony 3.0
+            if (isset($tag[0]['alias'])) {
+                $types[$tag[0]['alias']] = $serviceId;
+            } else {
+                $types[$serviceId] = $serviceId;
```

Comment:
I really don't want to add new features if they are not absolutely necessary. The Form component has way too many features already and is a nightmare to maintain. Everything that is not 100% necessary should be removed, if possible. This is one of these things.